MAINT: Enable stale bot in debug mode

connortann opened the pull request:
## Overview

- Implements stale bot as discussed in https://github.com/shap/shap/discussions/3051
- Updates maintainer docs with overview of issue triage and use of labels

I have suggested starting with "debug mode" so that no action is taken, but the output is written to the logs. I propose we start with a dry run, which requires merging into the master branch. Then if the logs look suitable then we can remove the debug option.

For more info about the action see https://github.com/actions/stale

## Discussion

This is related to #3423 , which addresses a slightly different situation. I think the two approaches could be complementary. I think this one is a good place to start, as the volume of "stale" issues is much higher than the "awaiting feedback" issues.

connortann committed:
Add stale bot in debug mode

connortann committed:
Markdown fixups

connortann committed:
Add docs on issue triage

connortann committed:
Better docs

connortann committed:
Fixup TOC

connortann committed:
Rewrap

CloseChoice commented on .github/workflows/stale_issues.yml:
```diff
@@ -0,0 +1,54 @@
+            We value your input! If you think this issue is still relevant, please
+            leave a comment to keep it open and remove the 'stale' label.
+            \n\n
+            If there's no activity in the next 30 days the issue will be closed."
```
```suggestion
            If there's no activity in the next 90 days the issue will be closed."
```

Or did I miss something? I thought line 24 configures this time span?!

connortann replied:
Ah good spot, thanks

CloseChoice commented on .github/workflows/stale_issues.yml:
```diff
@@ -0,0 +1,54 @@
+            Please leave a comment if you would still like the PR to remain open, and
+            the 'stale' label will be removed.
+            \n\n
+            Otherwise, if there is no activity in the next 30 days the PR will be
```
```suggestion
            Otherwise, if there is no activity in the next 90 days the PR will be
```

CloseChoice commented on .github/workflows/stale_issues.yml:
```diff
@@ -0,0 +1,54 @@
+          close-issue-message: "This issue has been automatically closed due to lack of
+            recent activity.
+            \n\n
+            Your input is important to us! Please feel free to open a new issue if the
```
why not just reopen the ticket?

connortann replied:
I believe than only maintainers can re-open issues, so most of the community will instead need to open a new one:

https://github.com/orgs/community/discussions/41926

CloseChoice commented:
Hmm, I am a bit torn on this one. On the one hand I am worried that we might lose valualbe information if we'll close tickets just because the person who opened it does not respond in time. But I would see this as totally fine procedure regarding PRs. On the other hand my former scrum master always said: "What's important comes back in no time" and I guess he's right about that. And I think you wrote good, clear and appreciative messages to the authors and are pretty defensive about the time frames. 

So I am fine with this change.

CloseChoice reviewed (approved):
Looks good to me, just some minor things to sort out

connortann committed:
Apply suggestions from code review

Co-authored-by: Tobias Pitters <31857876+CloseChoice@users.noreply.github.com>

connortann committed:
Add reference to docs and GH discussion

connortann commented:
Example run in debug mode (which is capped to one batch at a time):
https://github.com/shap/shap/actions/runs/7126747237/job/19405262741

```
Processed items: 388
├── Processed issues: 359
└── Processed PRs   : 29
New stale issues: 3
No longer stale items: 3
├── No longer stale issues: 1
└── No longer stale PRs   : 2
Deleted items labels: 3
├── Deleted issues labels: 1
└── Deleted PRs labels   : 2
Added issues labels: 3
Added issues comments: 3
Fetched items: 400
Fetched items events: 10
Fetched items comments: 10
Operations performed: 33
```
